operator/v1/types_ingress.go

```diff
@@ -512,6 +512,8 @@ const (
 
 // AWSLoadBalancerParameters provides configuration settings that are
 // specific to AWS load balancers.
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=SetEIPForNLBIngressController,rule="self.type == 'NLB' ? true : !has(self.networkLoadBalancer)",message="Network load balancer parameters are allowed only when load balancer type is NLB."
```

Comment:
Has this been discussed elsewhere? Unless I'm missing something, I don't think adding this CEL is a compatible change. Someone could have both classicLoadBalancer and networkLoadBalancer set, and this would cause their IngressController to be invalid on the next update/reconcile.

Reply:
@gcs278 Yes  @JoelSpeed suggested it here https://github.com/openshift/api/pull/1826#discussion_r1650776298

Reply:
But this is in a different file, on the IngressController. Didn't they mean for the Ingress object? Their comments were on `config/v1/types_ingress.go`.

Reply:
Yes I have made these changes only for config/v1 only to avoid compatibility issues